code/modules/mining/boulder_processing/boulder.dm

```diff
@@ -0,0 +1,279 @@
+		switch(picked.type)
+			if(/datum/material/iron)
+				cracked_ore = new /obj/item/stack/ore/iron(drop_location(), quantity)
+			if(/datum/material/gold)
+				cracked_ore = new /obj/item/stack/ore/gold(drop_location(), quantity)
+			if(/datum/material/silver)
+				cracked_ore = new /obj/item/stack/ore/silver(drop_location(), quantity)
+			if(/datum/material/plasma)
+				cracked_ore = new /obj/item/stack/ore/plasma(drop_location(), quantity)
+			if(/datum/material/diamond)
+				cracked_ore = new /obj/item/stack/ore/diamond(drop_location(), quantity)
+			if(/datum/material/glass)
+				cracked_ore = new /obj/item/stack/ore/glass/basalt(drop_location(), quantity)
+			if(/datum/material/bluespace)
+				cracked_ore = new /obj/item/stack/ore/bluespace_crystal(drop_location(), quantity)
+			if(/datum/material/titanium)
+				cracked_ore = new /obj/item/stack/ore/titanium(drop_location(), quantity)
+			if(/datum/material/uranium)
+				cracked_ore = new /obj/item/stack/ore/uranium(drop_location(), quantity)
+		SSblackbox.record_feedback("tally", "ore_mined", quantity, cracked_ore)
+
+/obj/item/boulder/proc/can_get_processed()
+	return COOLDOWN_FINISHED(src, processing_cooldown)
+
+/obj/item/boulder/proc/restart_processing_cooldown()
+	COOLDOWN_START(src, processing_cooldown, 2 SECONDS)
+
+/**
+ * Moves boulder contents to the drop location, and then deletes the boulder.
+ */
+/obj/item/boulder/proc/break_apart()
+	var/list/quips = list("Clang!", "Crack!", "Bang!", "Clunk!", "Clank!")
+	for(var/obj/item/content as anything in contents)
+		content.forceMove(get_turf(src))
+		visible_message(span_notice("[pick(quips)] Something falls out of \the [src]!"))
+		playsound(loc, 'sound/effects/picaxe1.ogg', 60, FALSE)
+	qdel(src)
+
+/**
+ * This is called when a boulder is spawned from a vent, and is used to set the boulder's icon as well as durability.
+ * We also set our boulder_size variable, which is used for inheiriting the icon_state later on if processed.
+ */
+/obj/item/boulder/proc/flavor_boulder(obj/structure/ore_vent/parent_vent, size = BOULDER_SIZE_SMALL, is_artifact = FALSE)
+	var/durability_min = size
+	var/durability_max = size + BOULDER_SIZE_SMALL
+	if(parent_vent)
+		durability_min = parent_vent.boulder_size
+		durability_max = parent_vent.boulder_size + BOULDER_SIZE_SMALL
+	durability = rand(durability_min, durability_max) //randomize durability a bit for some flavor.
+	boulder_size = size
+	if(parent_vent)
+		boulder_size = parent_vent.boulder_size
+		boulder_string = parent_vent.boulder_icon_state
+	update_appearance(UPDATE_ICON_STATE)
+
+/**
+ * Unique proc for gulag-style boulders, which adds a random amount of minerals to the boulder.
+ */
+/obj/item/boulder/proc/add_gulag_minerals(list_to_pick)
+	var/datum/material/new_material = pick_weight(gulag_minerals)
+	var/list/new_mats = list()
+	new_mats += new_material
```

Comment:
```suggestion
```

The next line assigns the element directly

Reply:
set_custom_materials specifically handles custom_material.on_applied, in order to apply materials specific effects like making plasma flamable for example, so there's some value in passing it through again.